text/0000-cargo-features2.md

```diff
@@ -0,0 +1,520 @@
+for the normal dependencies. As a build-dependency, it will have the `std`
+feature enabled. This means that `log` will be built twice, once without `std`
+and once with `std`.
+
+Note that a dependency shared between a build-dependency and proc-macro are
+still unified. This is intended to help reduce build times, and is expected to
+be unlikely to cause problems that feature unification usually cause because
+they are both being built for the host platform, and are only used at build
+time.
+
+## Resolver opt-in
+
+Testing has been performed on various projects. Some were found to fail to
+compile with the new resolver. This is because some dependencies are written
+to assume that features are enabled from another part of the graph. Because
+the new resolver results in a backwards-incompatible change in resolver
+behavior, the user must opt-in to use the new resolver. This can be done with
+the `resolver` field in `Cargo.toml`:
+
+```toml
+[package]
+name = "my-package"
+version = "1.0.0"
+resolver = "2"
+```
+
+Setting the resolver to `"2"` switches Cargo to use the new feature resolver.
+It also enables backwards-incompatible behavior detailed in [New command-line
+behavior](#new-command-line-behavior). `"2"` is the only valid option, if
+`resolver` is not specified then the old behavior is used.
+
+The `resolver` field is only honored in the top-level package or workspace, it
+is ignored in dependencies. This is because feature-unification is an
+inherently global decision.
+
+If using a virtual workspace, the root definition should be in the
+`[workspace]` table like this:
+
+```toml
+[workspace]
+members = ["member1", "member2"]
+resolver = "2"
+```
+
+For packages that encounter a problem due to missing feature declarations, it
+is backwards-compatible to add the missing features. Adding those missing
+features should not affect projects using the old resolver.
+
+It is intended that `resolver = "2"` will likely become the default setting in
+a future Rust Edition. See ["Default opt-in"](#default-opt-in) below for more
+details.
+
+## New command-line behavior
+
+The following changes are made to the behavior of selecting features on the
+command-line.
+
+* Features listed in the `--features` flag no longer pay attention to the
+  package in the current directory. Instead, it only enables the given
+  features for the selected packages. Additionally, the features are enabled
+  only if the the package defines the given features.
+
+  For example:
+
+      cargo build -p member1 -p member2 --features foo,bar
+
+  In this situation, features "foo" and "bar" are enabled on the given members
+  only if the member defines that feature. It is still an error if none of the
+  selected packages defines a given feature.
+
+* Features for individual packages can be enabled by using
+  `member_name/feature_name` syntax. For example, `cargo build --workspace
+  --feature member_name/feature_name` will build all packages in a workspace,
```

Comment:
I am assuming that we can do `cargo build -p member1 -p member2 --features member1/foo`. Would be a good idea expanding how usage in this item interacts with the usage in the above item.

Reply:
I'm not sure what kind of expansion you are looking for.  It enables the feature `foo` on the package `member1`.  You can also experiment with `-Zpackage-features` on nightly to see how it behaves.